amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/SchedulerTest.java

```diff
@@ -292,6 +297,39 @@ public final void testCleanupShardConsumers() {
         assertFalse(shardConsumer1.isShutdownRequested());
     }
 
+    @Test
+    public final void testParentShardDeletedAtShardEndHappyCase() {
```

Comment:
Also, can we add test case  when one child has started processing and other is at TRIM_HORIZON, the parent is not deleted

Reply:
See above comment